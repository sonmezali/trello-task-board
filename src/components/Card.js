import React from 'react'


const Card=(props)=> {
const handleDragStart=(e)=>{
    const target=e.target;
    e.dataTransfer.setData('card_id',target.id); 
}

const handleDrop=(e)=>{
     e.preventDefault();
     const cardId = e.dataTransfer.getData('card_id');    
     const draggedCard=props.cards.find(item=>item.id===cardId)   
     props.cards.splice(props.cards.findIndex(item=>item.id===cardId),1)
     props.cards.splice(props.cards.findIndex(item=>item.id===props.id),0,draggedCard)    
     props.setCards(props.cards)
     localStorage.setItem('cards',props.cards)
 }

// const handleDragOver=(e)=>{
//     e.stopPropagation()  
// }

    return (
        <div 
        index={props.index}
        id={props.id}
        draggable
        className={props.className}
        onDragStart={handleDragStart}
        // onDragOver={handleDragOver}
        onDrop={handleDrop}
        >
        {props.title}
            
        </div>
    )
}

export default Card;