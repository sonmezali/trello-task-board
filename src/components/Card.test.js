import React from 'react';
import Card from './Card';
import ReactDom, {unmountComponentAtNode} from 'react-dom';
import renderer from 'react-test-renderer';



describe('test Card component', () => {

    let container = null;
    beforeEach(() => {  
      container = document.createElement("div");
    });
  
    afterEach(() => {
      unmountComponentAtNode(container);
      container.remove();
      container = null;
    });
  
    test("renders without crashing", () => {
      const div = document.createElement("div");
      ReactDom.render(<Card/>, div);
    });
  
    test('matches snapshot', () => {
      const tree = renderer.create(<Card/>).toJSON();
      expect(tree).toMatchSnapshot();
    })
   
})