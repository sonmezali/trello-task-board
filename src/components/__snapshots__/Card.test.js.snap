// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Card component matches snapshot 1`] = `
<div
  draggable={true}
  onDragStart={[Function]}
  onDrop={[Function]}
/>
`;
